modules/exporter-common/src/main/scala/io/janstenpickle/trace4cats/export/ConfiguredHotswapSpanExporter.scala

```diff
@@ -0,0 +1,23 @@
+package io.janstenpickle.trace4cats.`export`
+
+import cats.effect.kernel.{Concurrent, Resource}
+import cats.kernel.Eq
+import io.janstenpickle.trace4cats.kernel.SpanExporter
+import io.janstenpickle.trace4cats.model.Batch
+
+trait ConfiguredHotswapSpanExporter[F[_], G[_], A] extends SpanExporter[F, G] {
+  def updateConfig(config: A): F[Boolean]
+}
+
+object ConfiguredHotswapSpanExporter {
+  def apply[F[_]: Concurrent, G[_], A: Eq](
+    initialConfig: A,
+    makeExporter: A => Resource[F, SpanExporter[F, G]]
+  ): Resource[F, ConfiguredHotswapSpanExporter[F, G, A]] =
+    HotswapSpanExporter[F, G, A](initialConfig, makeExporter(initialConfig)).map { underlying =>
+      new ConfiguredHotswapSpanExporter[F, G, A] {
+        override def updateConfig(config: A): F[Boolean] = underlying.update(config, makeExporter(config))
+        override def exportBatch(batch: Batch[G]): F[Unit] = underlying.exportBatch(batch)
+      }
+    }
+}
```

Comment:
This implementation pattern would probably be useful everywhere, I'm wondering if we make it the default for everything so that `A` and the underlying resources don't get out of sync when someone tries to use a low level implementation.

Reply:
This is a real boilerplate piece of code. And this may be abstracted over the resource.